pypika/queries.py

```diff
@@ -2042,6 +2058,77 @@ def __repr__(self) -> str:
         return self.__str__()
 
 
+class CreateIndexBuilder:
+
+    def __init__(self, dialect: Optional[Dialects] = None) -> None:
+        self._dialect = dialect
```

Comment:
is the `dialect` used somewhere?

Reply:
No, I reasoned that if someone needs to implement a vendor-specific SQL for index creation, that can be used in the future.

Reply:
I understand this reasoning but I believe it's better not to add it because it does not do anything currently. This can be added when the dialect specific SQL is added